### ISSUE TITLE
Support AssertJ's and Google Truth's fail() methods in RequireFailForTryCatchInJunitCheck

### ISSUE THREAD

**Issue by sebthom** [sevntu-checkstyle/sevntu.checkstyle#826]
The RequireFailForTryCatchInJunitCheck currently supports JUnit 4 `Assert.fail` and JUnit 5 `Assertions.fail` method.

We are currently migrating different projects to use AssertJ and Google Truth which provide similar methods:

```java
com.google.common.truth.Truth.assert_().fail();

org.assertj.core.api.Assertions.fail("message");

org.assertj.core.api.Assertions.failBecauseExceptionWasNotThrown(MyException.class);
```

RequireFailForTryCatchInJunitCheck should also support detect usage of these methods.

**Comment by romani** [comment]
fix is merged.
@sebthom , thank a lot for your time to make it happen.

**Comment by romani** [comment]
for this project we do not have defined plan of releases, we try to accumulate few fixes before release (release is not fully automated process).
Do you mind if I postpone release of your fix till end of March ?

**Comment by sebthom** [comment]
The early I can have the release the better. :-) Then I can finally get rid of the false positive warnings in my projects. But I can also wait till March.

**Comment by pbludov** [comment]
The case `assertWithMessage("...").fail()` should be supported too.
